playpen.py

```diff
@@ -1,21 +1,40 @@
 #!/usr/bin/env python3
 
 import subprocess
+import re, os.path
 
-def execute(version, command, arguments, data=None):
-    with subprocess.Popen(("playpen",
-                           "root-" + version,
+g_wrapper_cmd_cache = {}
+
+def execute(channel, command, arguments, data=None, skip_playpen=False, debug_me=False):
+    cache_key = (channel, command)
+    wrapper_cmd = g_wrapper_cmd_cache.get(cache_key, None)
+    if wrapper_cmd is None:
+        if skip_playpen:
+            wrapper_cmd = ()
+            if os.path.exists(command):
+                with open(command) as fp:
+                    if re.match(r'#!.*sh', fp.readline()):
+                        wrapper_cmd = ("sh", )  # compile.sh has non-standard #!/usr/bin/dash
```

Comment:
Yes it is (I'm on Debian.)

Arch is a weird distro. The [tar binary package of Dash](https://www.archlinux.org/packages/core/x86_64/dash/) contains `/usr/bin/dash` but its install/uninstall hook script tweaks `/etc/shells` as if it's at `/bin/dash`.  If there's really no `/bin/*sh` under an Arch chroot, I have to live with the above clumsiness.

The quickest solution is changing the shebang to `#!/usr/bin/env sh` but I think it is marginally more secure to keep using the current absolute shebang path.